### PR TITLE
Add AGUI support for adk

### PR DESCRIPTION
**Pull request by malithaYL**
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test update
- [ ] CI/CD update
- [ ] Other (please describe):

## Related Issues

Fixes #
Relates to #

## Changes Made

- 
- 
- 

## Testing

- [ ] Unit tests pass locally
- [ ] Integration tests pass locally
- [ ] Manual testing completed
- [ ] New tests added for changes

## Checklist

- [ ] My code follows the project's style guidelines
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by amithad — DISMISSED** [review]

File: ak-py/src/agentkernel/framework/adk/adk.py [hunk 1]
```diff
@@ -271,29 +293,126 @@ async def stream(self, agent: Any, session: Session, requests: list[AgentRequest
...
+                    elif message_id is not None:
+                        yield MessageEnd(message_id=message_id)
+                        message_id = None
+                    elif chunk:
+                        whole = uuid4().hex
+                        yield MessageStart(message_id=whole)
+                        yield TextDelta(message_id=whole, content=chunk)
+                        yield MessageEnd(message_id=whole)
+
+                    for tool_event in self._tool_events(event):
```
Comment (amithad) [inline ak-py/src/agentkernel/framework/adk/adk.py]: **[suggestion]** Tool-call events can be emitted inside an open reasoning trace, and a trace that resumes after the call keeps the same id.

- Reproduced against this branch: `partial(thought)` then `non-partial(function call, no answer text)` then `partial(thought)` yields `reasoning_start, reasoning_delta, tool_call_start, tool_call_args, tool_call_end, reasoning_delta, reasoning_end, ...` with one reasoning id throughout.
- Why it matters:
  - spec.md §10 (added in this PR) says "a trace resuming after a tool call is a second trace", but that only holds when answer text closed the trace first. A thinking model calling a tool straight out of reasoning, with no intervening answer text, is the common case.
  - OpenAI cannot produce this shape: `response.output_item.done` closes the reasoning item before the `function_call` item is added (openai.py maps those to `ReasoningEnd` / `ToolCallStart`), so the two adapters disagree on ordering for the same concept. This is the same cross-adapter divergence class the self-review fixed for text messages (plan.md iteration 5, step 6).
- Suggestion: close the open reasoning trace before emitting tool events (mirroring the message-boundary ordering fix), with a test confirmed failing first; alternatively record the divergence in §10 and soften the "second trace" sentence so the spec matches the code.

File: ak-py/src/agentkernel/framework/adk/adk.py [hunk 2]
```diff
@@ -271,29 +293,126 @@ async def stream(self, agent: Any, session: Session, requests: list[AgentRequest
...
-                        continue
-                    chunk = "".join(getattr(part, "text", "") or "" for part in event.content.parts)
-                    if chunk:
-                        yield chunk
-
-                # Only after the stream drains normally, so a disconnect or mid-stream error leaves the stored
-                # context intact. Deliberately not in a finally.
+                    chunk, thinking = self._event_text(event)
+
+                    if getattr(event, "partial", False) and thinking:
```
Comment (amithad) [inline ak-py/src/agentkernel/framework/adk/adk.py:307]: **[question]** Reasoning that arrives only on a non-partial event is silently dropped. Is that intended?

- A turn that never streamed got a whole-message fallback for its text (the `elif chunk:` branch below), but the same turn's `thought` parts are dropped: reproduced with a single non-partial event carrying thought + text, which emits only `message_start, text_delta, message_end`.
- spec.md §10 says the reasoning trace "opens on the first thought text" without restricting it to partials, and neither this docstring nor spec/plan records the asymmetry (the docstring documents the text fallback only).
- If deliberate (reasoning is auxiliary, so dropping it does not empty the reply the way dropping text would), a line here and in §10 would keep the next reader from "fixing" it; otherwise mirror the fallback with a whole `ReasoningStart` / `ReasoningDelta` / `ReasoningEnd` trace.

File: docs/specs/523-ag-ui-support/plan.md [hunk 3]
```diff
@@ -414,7 +414,63 @@ branch until PR 6 deletes it.
   3. Map function calls and responses onto the tool-call events.
   4. Update `tests/test_adk_runner.py`, including a test that two concurrent `stream()` calls on the
     same runner instance do not share a `message_id`.
-- **Verify:** `uv run pytest tests/test_adk_runner.py`, then the full suite.
+  5. Decisions and corrections taken while implementing:
+     - **`tests/test_tool_adk.py` also changes, and §10's test tables did not say so.** It holds two
```
Comment (amithad) [inline docs/specs/523-ag-ui-support/plan.md:418]: **[nit]** Iteration 5's **Files:** line (line 407, above this hunk) still lists only `framework/adk/adk.py` and `tests/test_adk_runner.py`. This bullet and the updated **Verify:** line now cover `tests/test_tool_adk.py` as well, so add it to the Files list too.

**Review by amithad — COMMENTED** [review]
Reviewed this PR's own delta against the stack base (`feature/523-agui-4-openai-langgraph`): the ADK adapter's event mapping, the two test files, and the spec.md/plan.md updates.

**Overall: a thorough, spec-conformant PR.** Every spec.md §10 ADK requirement is implemented: derived message boundaries kept as locals inside `stream` (with the concurrency test negative-tested), tool events keyed on `FunctionCall.id` with no-id entries skipped, reasoning split from answer text and kept out of the message stream, tool events emitted after message boundaries, and handoffs mapped as ordinary tool calls with an SDK-pinned name test. Deviations found while implementing are recorded in spec.md/plan.md rather than silently absorbed, which is exactly what the staged spec process asks for.

**Findings: 1 [suggestion], 1 [question], 1 [nit], all inline.**

Spec documents (reviewed before the code):
- The updated claims verify: `tests/test_tool_adk.py:677, 713` are the two stream-asserting tests on the base tree; `google/adk/flows/llm_flows/functions.py` does wrap a non-dict tool return in `{"result": ...}` in the pinned SDK; the corrected changing-tests table matches what this PR actually edits.
- One internal-consistency point in the added §10 text (the "a trace resuming after a tool call is a second trace" sentence) is raised inline on adk.py, anchored where the behavior lives.

Verification performed: extracted the PR head into an isolated tree and ran `tests/test_adk_runner.py tests/test_tool_adk.py` against it (93 passed); probed the two reasoning edge cases raised inline to confirm they are real rather than read off the diff.

Not anchorable to the diff:
- The PR template is entirely unfilled (empty description, empty "Fixes #", unchecked testing checklist). ak-dev-code-quality's PR guidelines ask for it, and linking issue #523 plus naming this as PR 5 of the stack would help reviewers landing here without context.

CI at review time: lint, CodeQL, build-ak-py, script-tests, and transport-integration-tests pass; unit-tests and the e2e matrix were still pending.

File: docs/specs/523-ag-ui-support/plan.md (hunk 3 above)
Comment (malithaYL) [reply docs/specs/523-ag-ui-support/plan.md:418]: Updated plan.md

File: ak-py/src/agentkernel/framework/adk/adk.py (hunk 2 above)
Comment (malithaYL) [reply ak-py/src/agentkernel/framework/adk/adk.py:307]: Not intended. It is fixed by mirroring the text fallback a thought that arrives only on a non-partial event is now emitted as a whole `ReasoningStart / ReasoningDelta / ReasoningEnd` trace.

**Review by amithad — DISMISSED** [review]

**Review by amithad — APPROVED** [review]

**Review by SandunYL — APPROVED** [review]
